### ISSUE TITLE
Feature Request: Workflow webhook flexibility

### ISSUE THREAD

**Issue by UFOSmuggler** [MISP/MISP#9321]
### Is your feature request related to a problem? Please describe.

The MISP workflow webhook feature is a bit limited and does not seem to allow a few simple cases:

1. Configurable HTTP request headers (ie 'Authorization: blah', also mentioned [here](https://github.com/MISP/MISP/issues/9231))
2. Configurable HTTP request type (ie POST, PUT, GET etc)
3. Configurable HTTP request body which also allows event data to be passed
4. Event data in the webhook target url (ie https://misp.example.com/events/publish/{Event.uuid})
5. Self signed certs

If I had this I could pretty much use it to undertake arbitrary tasks on the MISP instance firing the webhook vi its own REST API, which would be rad to the max.

### Describe the solution you'd like

Basically, the implementation of the above.  

Number 1 should allow an arbitrary number of headers, preferably allowing them to overwrite the existing headers on the request.
Where event data is used (like Event.uuid), multiple event datas should be passable to the input field, like the request body and webhook target url.

### Describe alternatives you've considered

An alternative would be a webhook reflector that receives the MISP webhook, and translates it into a MISP REST API request back to MISP.

### Additional context

This would make it easy to work without missing workflow outputs like the ability to publish events.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by Kagee** [comment]
Self-signed certs shooooould work if one follows https://github.com/MISP/MISP/wiki/Frequently-Asked-Questions#21-how-do-i-install-a-new-self-signedorganization-internal-to-the-misp-trust-store-how-do-i-update-what-root-cas-misp-trusts-how-do-i-make-misp-use-my-system-certificatetrust-store ?

**Comment by UFOSmuggler** [comment]
> Self-signed certs shooooould work if one follows https://github.com/MISP/MISP/wiki/Frequently-Asked-Questions#21-how-do-i-install-a-new-self-signedorganization-internal-to-the-misp-trust-store-how-do-i-update-what-root-cas-misp-trusts-how-do-i-make-misp-use-my-system-certificatetrust-store ?

true enough, but would still be nice to just have it as a checkbox in the webhook definition.

**Comment by mokaddem** [comment]
Hey @UFOSmuggler could you share with us what type of tasks you'd do by calling MISP's own REST API from a workflow? Maybe these could be `action` module?

**Comment by UFOSmuggler** [comment]
> Hey @UFOSmuggler could you share with us what type of tasks you'd do by calling MISP's own REST API from a workflow? Maybe these could be `action` module?

they definitely could be.  i mentioned publishing in my example, as missing workflow functionality.

by and large though, this additional webhook functionality could also drive other products (i have some ideas for taxii sidecar workflows, and interacting with lambdas on some of my other projects), so i don't think we should discount this feature request in favour of discrete action modules for things like publishing that will only benefit narrow use cases.

**Comment by mokaddem** [comment]
Absolutely agreed with the driving other applications. That's the main purpose of the webhook.
Although, I'd favor having built-in action module for operation that you'd perform using a self API call. So, if you have additional suggestion than a publish module, let us know!

**Comment by mokaddem** [comment]
@UFOSmuggler thanks again for the feedback!
05714df and 8e82260 should solve your issue. Let us know if it doesn't and I'll reopen the it.

**Comment by UFOSmuggler** [comment]
Thanks for all your work @mokaddem, this is great!

Apologies for not responding, have been a bit busy and only now have had some time to take a look.

I note that some things are not interpreted as event elements.

If I configure a webhook like this:

![Screenshot from 2023-10-25 11-05-06](https://github.com/MISP/MISP/assets/85923673/737efa1d-b4b6-4675-b749-a465e36c34f9)

I receive this:

```
127.0.0.1 - - [25/Oct/2023 00:02:56] "POST /{{Event.uuid}} HTTP/1.1" 200 -
{'Accept': 'application/json',
 'Accept-Encoding': 'br, gzip',
 'Connection': 'close',
 'Content-Length': '45',
 'Content-Type': 'application/json',
 'Event': '{{ Event.uuid }}',
 'Host': 'localhost:27051',
 'User-Agent': 'MISP 2.4.178 - #78c1d56eb4a61e79a6f9eecc71551990450988cf'}
'Event: ba62fe74-df34-41b5-8666-8fbdb77617fa'
```

Unless I'm holding it wrong, it appears the only time event elements are interpreted is in the request body.  Could this functionality also be implemented?

**Comment by mokaddem** [comment]
@UFOSmuggler This is indeed correct. Only the Payload was "jinja-aware". This has been fixed by c28e7876390561247d1d1a8f582a242fbff3f582 where you can now adapt the header and URL based on the roaming data.
I've added a small icon to show which field can hold a jinja template to avoid further confusion.
![image](https://github.com/MISP/MISP/assets/6977223/41921ae8-c564-4c57-943a-ebc8ecd33dd7)

Thanks again for the feedback!

**Comment by UFOSmuggler** [comment]
This works great, thanks again @mokaddem, this is excellent work!
